modal/stub.py

```diff
@@ -371,22 +368,15 @@ def _get_default_image(self):
     def _pty_input_stream(self):
         return self._blueprint.get("_pty_input_stream", None)
 
-    def _get_function_mounts(self, function_info: FunctionInfo):
-        # Get the common mounts for the stub.
-        mounts = list(self._mounts)
-
-        # Create client mount
-        if self._client_mount is None:
-            self._client_mount = _get_client_mount()
-        mounts.append(self._client_mount)
+    def _get_local_mounts(self):
+        all_mounts = [
+            *self._mounts,
+        ]
+        for function_handle in self.registered_functions.values():
+            function = self[function_handle._info.get_tag()]
```

Comment:
this is pretty janky but we can fix it when we merge providers and handles

maybe add a comment that this is ugly though

Reply:
Yeah it's pretty horrible 😢 And it's only used by `modal serve` in order to determine which mounts to watch for file changes... Adding a comment